website/js/app/app.js

```diff
@@ -0,0 +1,124 @@
+        // for deep links. So, alas, we need the anchor.
+        $locationProvider.html5Mode(false);
+    });
+
+    // Directive for the left-nav, the table of contents.
+    app.directive('cmdletExplorer', function() {
+        return {
+            restrict: 'E',
+            templateUrl: prefix + 'cmdlet-explorer.ng',
+            controller: 'CmdletExplorerController',
+            controllerAs: 'cmdletExplorerCtrl',
+            scope: {
+                documentationObj: '='
+            }
+        };
+    });
+
+    // Directive for rendering a PowerShell cmdlet's syntax and parameters.
+    app.directive('syntaxWidget', function() {
+        return {
+            restrict: 'E',
+            templateUrl: prefix + 'syntax-widget.ng',
+            controller: 'SyntaxWidgetController',
+            controllerAs: 'syntaxCtrl',
+            scope: {
+                syntax: '='
+            }
+        };
+    });
+
+    // Directive for rendering a PowerShell cmdlet's parameters.
+    app.directive('parametersWidget', function() {
+        return {
+            restrict: 'E',
+            templateUrl: prefix + 'parameters-widget.ng',
+            controller: 'ParametersWidgetController',
+            controllerAs: 'parametersCtrl',
+            scope: {
+                parameters: '='
+            }
+        };
+    });
+
+    // Filter to extract the simplified name from a full-qualified .NET type name. e.g.
+    // "System.String" to "String".
+    app.filter('stripNamespace', function() {
+        var genericListPattern = /(.+Generic.List`1\[\[)([^,]+)(, .*\]\])/g
+        return function(rawInput) {
+            var input = rawInput || '';
+
+            // HACK: Some cmdlets (Add-GceFirewall) accept Lists<T> for their parameters instead of
+            // simple arrays. This is a design flaw in the cmdlet. But we paper over them for now,
+            // until we clean up our cmdlets. For now we special case it.
+            if (input.indexOf('Generic.List`1') != -1) {
+                input = input.replace(genericListPattern, function(m, before, type, after) {
+                    return type + '[]';
+                });
+            }
+
+            // Location of the last namespace separator (.), nested type delimiter (+), or -1.
+            // e.g. System.String, Namespace.ParentType+NestedType.
+            var splitPoint = Math.max(
+                input.lastIndexOf('.'),
+                input.lastIndexOf('+'));
+            return input.substring(splitPoint + 1);
+        };
+    });
+
+    // Filter for rewriting raw text as HTML. To be passed to the ng-bind-html directive, which
+    // also takes care of sanitization.
+    // - Converts arrays of strings to separate <p> elements.
+    // - Puts quoted strings in <code class="text"> elements.
+    // - Puts references to cmdlets in <code class="cmdlet"> elements.
+    app.filter('applyHtmlStyling', function() {
```

Comment:
Unfortunately it can't. For cmdlet descriptions we regularly reference other cmdlets and included string literals. But it is just typed as a string.

We could "preprocess" those strings before we bind it on the view. e.g. "cmdletInfo.description" vs "cmdletInfo.descriptionInHtml", but we would still need to use ng-bind-html. At least with it being a filter it is easy to express. e.g.:

`<div ng-bind-html="contentCtrl.cmdletInfo.description | applyHtmlStyling"></div>`